addons/analytic/models/analytic_plan.py

```diff
@@ -290,6 +290,31 @@ def _sync_plan_column(self):
                     'index': True,
                 })
 
+    def write(self, vals):
+        if 'parent_id' in vals:
+            parent_id = vals['parent_id']
+            if parent_id:  # Update accounts in analytic lines before _sync_plan_column() unlinks child plan's column
+                parent_column_name = self.env['account.analytic.plan'].browse(parent_id)._column_name()
+                for plan in self:
+                    self.env['account.analytic.account']._update_accounts_in_analytic_lines(
+                        new_fname=parent_column_name,
+                        current_fname=plan._column_name(),
+                        account_ids=plan.account_ids,
+                    )
+            else:  # Store parents being removed, as accounts are updated after _sync_plan_column() creates the new column
+                parent_ids_map = {plan: plan.parent_id for plan in self}
```

Comment:
```suggestion
                plan2previous_parent = {plan: plan.parent_id for plan in self if plan.parent_id}
```

Don't try to change anything if it was already `False`